### ISSUE TITLE
Keep full url when switching version ?

### ISSUE THREAD

**Issue by nvuillam** [jimporter/mike#81]
Hi,

Thanks for the great tool !

Is there a way to remain on the current page when switching version ?

Example, when I'm on https://megalinter.github.io/v5.0.5/all_linters/ , I would like to land on https://megalinter.github.io/beta/all_linters/ when I switch to beta version, but unfortunately it always brings me back to the root URL

![image](https://user-images.githubusercontent.com/17500430/141271664-0766bb97-47ab-431d-8c5f-f84e2b68582b.png)

![image](https://user-images.githubusercontent.com/17500430/141271862-444c59e3-4d35-4f1b-ba29-2047e38ce766.png)

![image](https://user-images.githubusercontent.com/17500430/141272121-cd2cfe92-ea4f-4329-828f-2dce4fa31361.png)

Maybe with a javascript hack ? ( I already do that to rename/reorder versions in the dropdown widget) -> https://github.com/megalinter/megalinter/blob/main/docs/javascripts/version-mike.js

**Comment by squidfunk** [comment]
This feature is [already implemented](https://squidfunk.github.io/mkdocs-material/setup/setting-up-versioning/#stay-on-page) in Material for MkDocs, which you're using.

**Comment by jimporter** [comment]
Yeah, this is a goal for 2.0, but there are some subtleties I haven't quite worked out, like what to do when the directory structure has changed between versions. In that case, it'd be nice to redirect to the homepage for the specified version, or at least to have a nicer 404 page than we currently have.

**Comment by nvuillam** [comment]
> This feature is [already implemented](https://squidfunk.github.io/mkdocs-material/setup/setting-up-versioning/#stay-on-page) in Material for MkDocs, which you're using.

I saw that but it's only in insiders version for now ^^

**Comment by nvuillam** [comment]
> Yeah, this is a goal for 2.0, but there are some subtleties I haven't quite worked out, like what to do when the directory structure has changed between versions. In that case, it'd be nice to redirect to the homepage for the specified version, or at least to have a nicer 404 page than we currently have.

I agree,a nice 404 with a message "this page was not existing in version XXX" would be nice :)

**Comment by bridgeL** [comment]
> Maybe with a javascript hack ? ( I already do that to rename/reorder versions in the dropdown widget) -> [megalinter/megalinter@`main`/docs/javascripts/version-mike.js](https://github.com/megalinter/megalinter/blob/main/docs/javascripts/version-mike.js?rgh-link-date=2021-11-11T09%3A21%3A54Z)

I have tried your js code, but it seems that it doesn't work, maybe I have got something wrong. I write an another js code and it can work. Just put it here for someone who may need it.

```js
(window.onload = function () {
  var pathname = window.location.pathname.replace(/^\//, "");
  pathname = pathname.slice(pathname.indexOf("/") + 1);

  setInterval(function () {
    var items = document.querySelectorAll('.md-version__link:not(.changed)');
    for (var item of items) {
      item.href += pathname;
      item.className += " changed";
      console.log(item.innerHTML, "change successfully");
    }
  }, 100);
})();
```

**Comment by ilyagr** [comment]
The way `mkdocs-material` seems to do it is to fetch `sitemap.xml` to figure out which URLs are valid for each version. See

https://github.com/squidfunk/mkdocs-material/blob/f022873c69bb10b419dfbecab2cc0a2d6ac794b1/src/assets/javascripts/integrations/version/index.ts#L132

and

https://github.com/squidfunk/mkdocs-material/blob/f022873c69bb10b419dfbecab2cc0a2d6ac794b1/src/assets/javascripts/integrations/sitemap/index.ts#L91

I was briefly considering trying to implement this for `readthedocs` theme, but I suspect that my javascript knowledge is not up to par, especially if we want the code to work in many browsers.

**Comment by jimporter** [comment]
> The way `mkdocs-material` seems to do it is to fetch `sitemap.xml` to figure out which URLs are valid for each version.

Using `sitemap.xml` is an interesting idea. I think that would resolve a lot of the issues I was seeing with an implementation here. I'll look into it.
